src/mapValues.test.ts

```diff
@@ -34,3 +34,27 @@ describe('data last', () => {
     });
   });
 });
+
+describe('mapValues key types', () => {
+  test('interface', () => {
+    mapValues({} as { foo: unknown; bar: unknown }, (_, key) =>
+      expectTypeOf(key).toEqualTypeOf<'foo' | 'bar'>()
+    );
+  });
+
+  test('mapped type', () => {
+    mapValues({} as { [K in string]: unknown }, (_, key) => {
+      expectTypeOf(key).toEqualTypeOf<string>();
+    });
+  });
+
+  test('indexed signature', () => {
+    mapValues(
+      // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style -- must be "indexed signature"
+      {} as { [key: string]: unknown },
+      (_, key) => {
+        expectTypeOf(key).toEqualTypeOf<string>();
+      }
+    );
+  });
```

Comment:
Just to make sure we cover all our bases, can you add tests for:
* `{ [key in number]: unknown }`
* `{ [key: number]: unknown }`
* `{ [key in `prefix${string}`]: unknown }`
* `{ [key: `prefix${string}`]: unknown }`

Reply:
good, also add tests for symbols